src/main/kotlin/io/lettuce/core/api/coroutines/RedisKeyCoroutinesCommandsExtensions.kt

```diff
@@ -0,0 +1,27 @@
+package io.lettuce.core.api.coroutines
+
+import io.lettuce.core.ScanArgs
+import io.lettuce.core.ScanStream
+import io.lettuce.core.cluster.api.coroutines.RedisClusterCoroutinesCommandsImpl
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.reactive.asFlow
+
+/**
+ * Sequentially iterate the keys space.
+ *
+ * @param scanArgs scan arguments.
+ * @return `Flow<K>` flow of keys.
+ */
+@JvmOverloads
+fun <K : Any, V : Any> RedisKeyCoroutinesCommands<K, V>.scan(scanArgs: ScanArgs? = null): Flow<K> {
```

Comment:
Overloading the keys commands for Coroutines uses a different design than the other API (sync, reactive) where `ScanStream`/`ScanIterator` are separate and the API package doesn't depend on `ScanStream`/`ScanIterator`.

Let's keep it the same style and add only an extension for `ScanStream` taking `RedisKeyCoroutinesCommands` as an argument and returning `Flow`.

Reply:
Adding extensions for `ScanStream` is useless and impossible in the right way for now.
https://youtrack.jetbrains.com/issue/KT-11968.
I propose to introduce `ScanFlow` singleton to access this API.

Reply:
Works for me, too. Feel free to do it that way.

Reply:
Done